src/bcccp/carpark/Carpark.java

```diff
@@ -1,150 +1,170 @@
-	}
+    @Override
+    public void recordAdhocTicketEntry() {
 
+        numberOfCarsParked++;
 
+    }
 
-	@Override
-	public void recordAdhocTicketExit() {
-		// TODO Auto-generated method stub
-		
-	}
 
+    @Override
+    public IAdhocTicket getAdhocTicket(String barcode) {
 
+        return adhocTicketDAO.findTicketByBarcode(barcode);
+    }
 
-	@Override
-	public void registerSeasonTicket(ISeasonTicket seasonTicket) {
-		// TODO Auto-generated method stub
-		
-	}
 
+    @Override
+    public float calculateAddHocTicketCharge(long entryDateTime) {
+        // Calculation: get current Date and Time, subtract entryDateTime, multiply result by
+        // the $ charge rate and return charge. Assumption that rates are a fixed rate of $5.00 per hour.
+        // Convert time to hours by dividing by 60,000. Assumption this is short-stay tariff car park.
 
+        float rates = 5.0f;
 
-	@Override
-	public void deregisterSeasonTicket(ISeasonTicket seasonTicket) {
-		// TODO Auto-generated method stub
-		
-	}
+        Date dateTime = new Date();
 
+        float chargeAmount = (dateTime.getTime() - entryDateTime) * rates / 60000;
 
+        return chargeAmount;
+    }
 
-	@Override
-	public boolean isSeasonTicketValid(String ticketId) {
-		// TODO Auto-generated method stub
-		return false;
-	}
 
+    @Override
+    public void recordAdhocTicketExit() {
 
+        numberOfCarsParked--;
 
-	@Override
-	public boolean isSeasonTicketInUse(String ticketId) {
-		// TODO Auto-generated method stub
-		return false;
-	}
+    }
 
 
+    @Override
+    public void registerSeasonTicket(ISeasonTicket seasonTicket) {
 
-	@Override
-	public void recordSeasonTicketEntry(String ticketId) {
-		// TODO Auto-generated method stub
-		
-	}
+        seasonTicketDAO.registerTicket(seasonTicket);
 
+    }
 
 
-	@Override
-	public void recordSeasonTicketExit(String ticketId) {
-		// TODO Auto-generated method stub
-		
-	}
+    @Override
+    public void deregisterSeasonTicket(ISeasonTicket seasonTicket) {
+
+        seasonTicketDAO.deregisterTicket(seasonTicket);
+
+    }
+
+
+    @Override
+    public boolean isSeasonTicketValid(String ticketId) {
+
+        // If today's date is within the startValidPeriod and endValidPeriod, the season ticket is valid
+
+        Date dateTime = new Date();
+
+        ISeasonTicket sTicket = seasonTicketDAO.findTicketById(ticketId);
+
+        return ((dateTime.getTime() >= sTicket.getStartValidPeriod()) &&
+                (dateTime.getTime() <= sTicket.getEndValidPeriod()))
+                ? true : false;
+    }
+
+
+    @Override
+    public boolean isSeasonTicketInUse(String ticketId) {
+
+        ISeasonTicket sTicket = seasonTicketDAO.findTicketById(ticketId);
+
+        return sTicket.inUse();
+    }
+
+
+    @Override
+    public void recordSeasonTicketEntry(String ticketId) {
+
+        seasonTicketDAO.recordTicketEntry(ticketId);
+
+    }
+
+
+    @Override
+    public void recordSeasonTicketExit(String ticketId) {
```

Comment:
JAVADOC: Please add javadoc as per Google Java Style guide.
At the minimum, Javadoc is present for every public class, and every public or protected member of such a class.